modin/core/storage_formats/base/query_compiler.py

```diff
@@ -3079,9 +3079,21 @@ def sort_rows_by_column_values(
         BaseQueryCompiler
             New QueryCompiler that contains result of the sort.
         """
-        return DataFrameDefault.register(pandas.DataFrame.sort_values)(
+        # Avoid index/column name collisions by renaming and restoring after sorting
+        index_renaming = None
+        if not isinstance(columns, list):
+            columns = [columns]
+        if any(name in columns for name in self.index.names):
```

Comment:
This check should be more relaxed since this is on the frontend API layer, and it should still technically work if a user passes in a set/tuple or other collection object as an argument. Try something like this instead:

```suggestion
        if not is_scalar(columns) and any(name in columns for name in self.index.names):
```